curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641da42481d90c4314c99e94.md

```diff
@@ -0,0 +1,170 @@
+---
+id: 641da42481d90c4314c99e94
+title: Step 3
+challengeType: 0
+dashedName: step-3
+---
+
+# --description--
+
+The `fetch()` method returns a `Promise`, which is a placeholder object that will either be fulfilled if your request is successful, or rejected if your request is unsuccessful.
+
+If the `Promise` is fulfilled, it resolves to a `Response` object, and you can use the `.then()` method to access the `Response`.
+
+Here's how you can chain `.then()` to the `fetch()` method:
+
+```js
+fetch("sample-url-goes-here")
+  .then((res) => console.log(res))
+```
+
+Chain the `.then()` method to your `fetch` call. Inside the `.then()` method, add a callback function with a `res` parameter, then log the `res` to the console so you can see the `Response` object. 
+
+Open your console and expand the `Response` object to see what it contains. Again, don't terminate the code with a semi-colon yet.
+
+**Note**: You can open up the built-in freeCodeCamp console or your browser's console to see the data you're getting.
+
+# --hints--
+
+You should use the `fetch()` method to make a `GET` request to `https://cdn.freecodecamp.org/curriculum/news-author-page/authors.json`.
+
+```js
+assert.match(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\)/)
+```
+
+You should chain the `.then()` method to your `fetch()` syntax
+
+```js
+assert.match(code, /\.then\(/)
+```
+
+You should add a `res` parameter to your `.then()` method.
+
+```js
+assert.match(code, /\.then\(\(?res\)?/)
+```
+
+Your `.then()` method should use arrow function syntax.
+
+```js
+assert.match(code, /\.then\(\(?res\)?\s*=>\s*/)
+```
+
+You should log the `res` parameter to the console so you can see the `Response` object.
+
+```js
+assert.match(code, /\.then\(\(?res\)?\s*=>\s*console\.log\(res\)\)/)
+```
```

Comment:
````suggestion
You should chain the `.then()` method to your `fetch()` syntax

```js
assert.match(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\)\s*\.then\(/)
```

You should add a `res` parameter to your `.then()` method.

```js
assert.match(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\)\s*\\.then\(\(?res\)?/)
```

Your `.then()` method should use arrow function syntax.

```js
assert.match(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\)\s*\\.then\(\(?res\)?\s*=>\s*/)
```

You should log the `res` parameter to the console so you can see the `Response` object.

```js
assert.match(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\)\s*\\.then\(\(?res\)?\s*=>\s*console\.log\(res\)\)/)
```
````

We can test that the camper actually chained it.